lib/swoosh/adapters/mua.ex

```diff
@@ -211,8 +211,27 @@ defmodule Swoosh.Adapters.Mua do
   end
 
   defp put_headers(mail, headers) do
+    keys = headers |> Map.keys() |> Enum.map(&String.downcase/1)
+
+    has_date? = "date" in keys
+    has_message_id? = "message-id" in keys
+
+    headers = if has_date?, do: headers, else: Map.put(headers, "Date", DateTime.utc_now())
+    headers = if has_message_id?, do: headers, else: Map.put(headers, "Message-ID", message_id())
+
     Enum.reduce(headers, mail, fn {key, value}, mail ->
       Mail.Message.put_header(mail, key, value)
     end)
   end
+
+  defp message_id do
+    Base.hex_encode32(
+      <<
+        System.system_time(:nanosecond)::64,
+        :erlang.phash2({node(), self()}, 16_777_216)::24,
+        :erlang.unique_integer()::32
+      >>,
+      case: :lower
+    )
```

Comment:
I copied this from [Joken](https://github.com/joken-elixir/joken/blob/de299190741a7991dd2e66da98ec3f9d91fda737/lib/joken.ex#L190-L204) which itself is inspired by [Plug.RequestId](https://github.com/elixir-plug/plug/blob/v1.16.1/lib/plug/request_id.ex#L81-L89) 

The generated IDs that look like this:

```
2vm6j5a0ddqhnr8pdc000032
2vm6j5a0dg049r8pdc000042
2vm6j5a0dick7r8pdc000052
2vm6j5a0dililr8pdc000062
2vm6j5a0dithrr8pdc000072
2vm6j5a0dj6lhr8pdc000082
2vm6j5a0djditr8pdc000092
2vm6j5a0dk0k9r8pdc0000a2
2vm6j5a0dk8e9r8pdc0000b2
2vm6j5a0dkgntr8pdc0000c2
```